simpeg/optimization.py

```diff
@@ -1196,88 +1339,127 @@ def evalFunction(x, return_g=False):
         return x
 
 
-class ProjectedGNCG(BFGS, Minimize, Remember):
-    def __init__(self, **kwargs):
-        Minimize.__init__(self, **kwargs)
+class ProjectedGNCG(Bounded, InexactGaussNewton):
+    def __init__(
+        self,
+        *,
+        lower=-np.inf,
+        upper=np.inf,
+        cg_maxiter=5,
+        cg_rtol=None,
+        cg_atol=None,
+        step_active_set=True,
+        active_set_grad_scale=1e-2,
+        **kwargs,
+    ):
+        if cg_rtol is None and cg_atol is None:
+            # Note these defaults match previous settings...
+            # but they're not good in general...
+            # Ideally they will change to cg_rtol=1E-3 and cg_atol=0.0
+            warnings.warn(
+                "The defaults for ProjectedGNCG will change in SimPEG 0.26.0. If you want to maintain the "
+                "previous behavoir, explicitly set 'cg_atol=1E-3' and 'cg_rtol=0.0'.",
+                FutureWarning,
+                stacklevel=2,
+            )
+            cg_atol = 1e-3
+            cg_rtol = 0.0
+        super().__init__(
+            lower=lower,
+            upper=upper,
+            cg_maxiter=cg_maxiter,
+            cg_rtol=cg_rtol,
+            cg_atol=cg_atol,
+            **kwargs,
+        )
 
-    name = "Projected GNCG"
+        self.step_active_set = step_active_set
+        self.active_set_grad_scale = active_set_grad_scale
 
-    maxIterCG = 5
-    tolCG = 1e-1
-    cg_count = 0
-    stepOffBoundsFact = 1e-2  # perturbation of the inactive set off the bounds
-    stepActiveset = True
-    lower = -np.inf
-    upper = np.inf
+        # initialize some tracking parameters
+        self.cg_count = 0
+        self.cg_abs_resid = np.inf
+        self.cg_rel_resid = np.inf
 
-    def _startup(self, x0):
-        # ensure bound vectors are the same size as the model
-        if not isinstance(self.lower, np.ndarray):
-            self.lower = np.ones_like(x0) * self.lower
-        if not isinstance(self.upper, np.ndarray):
-            self.upper = np.ones_like(x0) * self.upper
+        self.printers.extend(
+            [
+                IterationPrinters.iterationCG,
+                IterationPrinters.iteration_CG_rel_residual,
+                IterationPrinters.iteration_CG_abs_residual,
+            ]
+        )
 
-    @count
-    def projection(self, x):
```

Comment:
All functionality for the box bound constrains is moved to the `Bounded` mixin, which is also re-used by `ProjectedGradient`